# Keep `nix-util` and `nix-store` out of the public `Requires` field in
# `nix-store-c.pc`. Static consumers of the C API should only expose
# `nix-util-c` publicly and carry the C++ implementation libraries privately.

diff --git a/src/libstore-c/meson.build b/src/libstore-c/meson.build
--- a/src/libstore-c/meson.build
+++ b/src/libstore-c/meson.build
@@ -59,5 +59,25 @@ install_headers(headers, preserve_path : true)

 libraries_private = []

+if host_machine.system() != 'windows'
+  meson.add_install_script('sh', '-c', '''
+    pc="$MESON_INSTALL_DESTDIR_PREFIX/@0@"
+    tmp="$pc.tmp"
+    : > "$tmp"
+    while IFS= read -r line; do
+      case "$line" in
+        'Requires: nix-util-c, nix-util, nix-store')
+          printf '%s\n' 'Requires: nix-util-c' >> "$tmp"
+          printf '%s\n' 'Requires.private: nix-util, nix-store' >> "$tmp"
+          ;;
+        *)
+          printf '%s\n' "$line" >> "$tmp"
+          ;;
+      esac
+    done < "$pc"
+    mv "$tmp" "$pc"
+  '''.format(get_option('libdir') / 'pkgconfig' / 'nix-store-c.pc'))
+endif
+
 subdir('nix-meson-build-support/export')
